### PR TITLE
Unique ID: UUID

### LINKED ISSUE #3452: UUID
**Is your feature request related to a problem? Please describe.**
add UUID

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by jackwener**
#### What type of PR is this?
- [ ] bug
- [x] feature
- [ ] enhancement

#### What does this PR do?

Add UUID

#### Which issue(s)/PR(s) this PR relates to?

close #3452 
  
#### Special notes for your reviewer, ex. impact of this fix, etc:

None

#### Additional context:

None

#### Checklist：
- [ ] Documentation affected （Please add the label if documentation needs to be modified.)
- [ ] Incompatible （If it is incompatible, please describe it and add corresponding label.）
- [ ] Need to cherry-pick （If need to cherry-pick to some branches, please label the destination version(s).)
- [x] Performance impacted: Consumes more CPU/Memory

#### Release notes：
Please confirm whether to reflect in release notes and how to describe:
>

### CODE REVIEW

File: src/common/id/UUID.h [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+#include <boost/uuid/uuid.hpp>             // uuid class
+#include <boost/uuid/uuid_generators.hpp>  // generators
+#include <boost/uuid/uuid_io.hpp>          // streaming operators etc
+
+namespace nebula {
+
+class UUID {
+ public:
+  static boost::uuids::uuid getId() {
+    boost::uuids::uuid uuid = boost::uuids::random_generator()();
```
Comment (Shylock-Hg) [inline src/common/id/UUID.h:15]: As I known, the random generated uuid contains probability of collision. Do you consider it?

File: src/common/id/UUID.h [hunk 2]
```diff
@@ -0,0 +1,28 @@
+/* Copyright (c) 2021 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License.
+ */
+
+#include <boost/uuid/uuid.hpp>             // uuid class
+#include <boost/uuid/uuid_generators.hpp>  // generators
+#include <boost/uuid/uuid_io.hpp>          // streaming operators etc
+
```
Comment (jievince) [inline src/common/id/UUID.h]: delete them.

File: src/common/id/UUID.h [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+ * UUID v2: DCE security
+ * UUID v3: name-based MD5
+ * UUID v4: random
+ * UUID v5: name-based SHA-1
+ * boost implements uuid v4 and v5(is deprecated due to security concerns)
+ */
+namespace nebula {
+
+// the probability to find a duplicate within version-4 UUIDs is one in a billion.
+// the computed is in: https://en.wikipedia.org/wiki/Universally_unique_identifier
```
Comment (jievince) [inline src/common/id/UUID.h:21]: Improve the notes.

**Review by CPWstatic — APPROVED** [review]

**Review by jievince — APPROVED** [review]

**Comment by jackwener** [comment]
The reason why the v4 implementation of boost is used is because v1 v3 v5 have security issues and are deprecated.
v2 is basically rarely used.

**Pull request merged**